lib/lib-storage/src/Upload.ts

```diff
@@ -427,6 +437,28 @@ export class Upload extends EventEmitter {
     });
   }
 
+  private __validateUploadPart(dataPart: RawDataPart): void {
+    const actualPartSize = byteLength(dataPart.data) || undefined;
+
+    // Skip validation for single-part uploads (PUT operations)
+    if (dataPart.partNumber === 1 && dataPart.lastPart) {
+      return;
```

Comment:
it seems you could return here without checking the `actualPartSize` (move it after this)